src/test/regress/expected/failure_create_reference_table.out

```diff
@@ -0,0 +1,261 @@
+--
+-- Failure tests for creating reference table
+--
+CREATE SCHEMA failure_reference_table;
+SET search_path TO 'failure_reference_table';
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+CREATE TABLE ref_table(id int);
+INSERT INTO ref_table VALUES(1),(2),(3);
+-- Kill on sending first query to worker node, should error
+-- out and not create any placement
+SELECT citus.mitmproxy('conn.onQuery().kill()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT create_reference_table('ref_table');
+ERROR:  server closed the connection unexpectedly
+	This probably means the server terminated abnormally
+	before or while processing the request.
+CONTEXT:  while executing command on localhost:57640
+SELECT count(*) FROM pg_dist_shard_placement;
+ count 
+-------
+     0
+(1 row)
+
+-- Kill after creating transaction on worker node 
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT citus.mitmproxy('conn.onCommandComplete(command="BEGIN").kill()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT create_reference_table('ref_table');
+WARNING:  connection not open
+CONTEXT:  while executing command on localhost:57640
+ERROR:  connection error: localhost:57640
+DETAIL:  connection not open
+SELECT count(*) FROM pg_dist_shard_placement;
+ count 
+-------
+     0
+(1 row)
+
+-- Cancel after creating transaction on worker node
+SELECT citus.mitmproxy('conn.allow()');
```

Comment:
You don't need to do `conn.allow()` if you're going to immediately call `citus.mitmproxy` again. Every time you call `citus.mitmpeoxy()` the proxy re-configures itself and throws away the previous configuration.